csp_adapter_symphony/adapter.py

```diff
@@ -285,22 +302,28 @@ def _send_messages(
     """Read messages from msg_queue and write to symphony. msg_queue to contain instances of SymphonyMessage, or None to shut down"""
     send_message = config.get_retry_decorator()(send_symphony_message)
     message_create_url = config.message_create_url
+    create_im = config.get_retry_decorator()(create_im_stream)
+    im_create_url = config.im_create_url
     header = config.header
 
-    while True:
-        msg = msg_queue.get()
-        msg_queue.task_done()
-        if not msg:  # send None to kill
-            break
-
-        room_id = room_mapper.get_room_id(msg.room)
+    def _send_message(
+        msg: SymphonyMessage,
+    ):
+        if msg.room == "IM" and hasattr(msg, "user_id"):
+            stream_id = create_im(msg.user_id, im_create_url, header)
```

Comment:
thinking ahead to the [bot framework](https://github.com/point72/csp-bot), we should consider disallowing IMs at the config level. Seems reasonable that in a plugin-oriented setup, one might want to disallow IMs globally.

Reply:
it will be made public later today